packages/opentelemetry-plugin-user-interaction/src/userInteraction.ts

```diff
@@ -0,0 +1,452 @@
+
+import * as shimmer from 'shimmer';
+import { BasePlugin, hrTime, isWrapped } from '@opentelemetry/core';
+import * as types from '@opentelemetry/types';
+import { getElementXPath } from '@opentelemetry/web';
+import {
+  AsyncTask,
+  RunTaskFunction,
+  SpanData,
+  WindowWithZone,
+  ZoneTypeWithPrototype,
+} from './types';
+import { AttributeNames } from './enums/AttributeNames';
+import { VERSION } from './version';
+
+const ZONE_SCOPE_KEY = 'OT_ZONE_SCOPE';
+const EVENT_CLICK_NAME = 'event_click:';
+const EVENT_NAVIGATION_NAME = 'Navigation:';
+
+/**
+ * This class represents a UserInteraction plugin for auto instrumentation.
+ * If zone.js is available then it patches the zone otherwise it patches
+ * addEventListener of HTMLElement
+ */
+export class UserInteractionPlugin extends BasePlugin<unknown> {
+  readonly component: string = 'user-interaction';
+  readonly version = VERSION;
+  moduleName = this.component;
+  private _spansData = new WeakMap<types.Span, SpanData>();
+  private _zonePatched = false;
+
+  /**
+   * This will check if last task was timeout and will save the time to
+   * fix the user interaction when nothing happens
+   * This timeout comes from xhr plugin which is needed to collect information
+   * about last xhr main request from observer
+   * @param task
+   * @param span
+   */
+  private _checkForTimeout(task: AsyncTask, span: types.Span) {
+    const spanData = this._spansData.get(span);
+    if (spanData) {
+      if (task.source === 'setTimeout') {
+        spanData.hrTimeLastTimeout = hrTime();
+      } else if (
+        task.source !== 'Promise.then' &&
+        task.source !== 'setTimeout'
+      ) {
+        spanData.hrTimeLastTimeout = undefined;
+      }
+    }
+  }
+
+  /**
+   * Creates a new span
+   * @param element
+   * @param eventName
+   */
+  private _createSpan(
+    element: HTMLElement,
+    eventName: string
+  ): types.Span | undefined {
+    if (!element.getAttribute) {
+      return undefined;
+    }
+    if (element.hasAttribute('disabled')) {
+      return undefined;
+    }
+    const xpath = getElementXPath(element, true);
+    try {
+      const span = this._tracer.startSpan(`${EVENT_CLICK_NAME} ${xpath}`, {
+        attributes: {
+          [AttributeNames.COMPONENT]: this.component,
+          [AttributeNames.EVENT_TYPE]: eventName,
+          [AttributeNames.TARGET_ELEMENT]: element.tagName,
+          [AttributeNames.TARGET_XPATH]: xpath,
+          [AttributeNames.HTTP_URL]: window.location.href,
+          [AttributeNames.HTTP_USER_AGENT]: navigator.userAgent,
+        },
+        parent: this._tracer.getCurrentSpan(),
+        links: [],
+      });
+
+      this._spansData.set(span, {
+        taskCount: 0,
+      });
+
+      return span;
+    } catch (e) {
+      this._logger.error(this.component, e);
+    }
+    return undefined;
+  }
+
+  /**
+   * Decrement number of tasks that left in zone,
+   *     This is needed to be able to end span when no more tasks left
```

Comment:
Could you please remove extra indent, I think this is only applicable in case of `@param` when you show continued line.